packages/appstore/src/components/main-title-bar/asset-summary.tsx

```diff
@@ -1,71 +1,58 @@
+import { observer, useStore } from '@deriv/stores';
 import './asset-summary.scss';
 import TotalAssetsLoader from 'Components/pre-loader/total-assets-loader';
+import {
+    useTotalAccountBalance,
+    usePlatformRealAccounts,
+    usePlatformDemoAccount,
+    useCurrencyExchangeRate,
+    useCFDRealAccounts,
+    useCFDDemoAccounts,
+} from '@deriv/hooks';
 
-const AssetSummary = () => {
-    const { traders_hub, client, common } = useStores();
-    const {
-        selected_account_type,
-        platform_real_balance,
-        cfd_demo_balance,
-        platform_demo_balance,
-        cfd_real_balance,
-        is_eu_user,
-        no_CR_account,
-        no_MF_account,
-    } = traders_hub;
-    const { is_logging_in, is_switching } = client;
-    const { getExchangeRate } = common;
-
-    const [exchanged_rate_cfd_real, setExchangedRateCfdReal] = React.useState(1);
-    const [exchanged_rate_cfd_demo, setExchangedRateCfdDemo] = React.useState(1);
+const AssetSummary = observer(() => {
+    const { traders_hub, client } = useStore();
+    const { selected_account_type, is_eu_user, no_CR_account, no_MF_account } = traders_hub;
+    const { is_logging_in, is_switching, default_currency } = client;
 
-    React.useEffect(() => {
-        const getCurrentExchangeRate = (
-            currency: string,
-            setExchangeRate: React.Dispatch<React.SetStateAction<number>>,
-            base_currency = platform_real_balance.currency
-        ) => {
-            if (currency) {
-                getExchangeRate(currency, base_currency).then((res: number) => {
-                    setExchangeRate(res);
-                });
-            }
-        };
+    const platform_real_accounts = usePlatformRealAccounts();
+    const platform_demo_account = usePlatformDemoAccount();
+    const cfd_real_accounts = useCFDRealAccounts();
+    const cfd_demo_accounts = useCFDDemoAccounts();
+    const platform_real_balance = useTotalAccountBalance(platform_real_accounts);
+    const cfd_real_balance = useTotalAccountBalance(cfd_real_accounts);
+    const cfd_demo_balance = useTotalAccountBalance(cfd_demo_accounts);
+    const cfd_real_rate = useCurrencyExchangeRate(platform_real_balance.currency);
+    const cfd_demo_rate = useCurrencyExchangeRate(platform_demo_account?.currency || default_currency);
 
-        if (cfd_real_balance.currency !== platform_real_balance.currency) {
-            getCurrentExchangeRate(cfd_real_balance.currency, setExchangedRateCfdReal);
-        }
-        if (cfd_demo_balance.currency !== platform_demo_balance.currency) {
-            getCurrentExchangeRate(cfd_demo_balance.currency, setExchangedRateCfdDemo, platform_demo_balance.currency);
-        }
-    }, [
-        cfd_demo_balance.currency,
-        cfd_real_balance.currency,
-        getExchangeRate,
-        platform_demo_balance.currency,
-        platform_real_balance.currency,
-    ]);
-
-    const getTotalBalance = () => {
+    const totalBalance = useMemo(() => {
```

Comment:
```suggestion
    const is_real = selected_account_type === 'real';
    const real_total_balance = platform_real_balance.balance + cfd_real_balance.balance * cfd_real_rate;
    const demo_total_balance = (platform_demo_account?.balance || 0) + cfd_demo_balance.balance * cfd_demo_rate;
```
@shayan-deriv Use derived state instead.